libraries/botbuilder-repo-utils/src/updateVersions.ts

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as R from 'remeda';
+import minimist from 'minimist';
+import moment from 'moment';
+import path from 'path';
+import { Package } from './package';
+import { collectWorkspacePackages } from './workspace';
+import { failure, isFailure, Result, run, success } from './run';
+import { gitRoot, gitSha } from './git';
+import { readJsonFile, writeJsonFile } from './file';
+
+const GIT_SHA_LENGTH = 12;
+
+/**
+ * Computes the final version identifier components for a package.
+ * @param pkg package.json data
+ * @param newVersion new version requested
+ * @param options options to control versioning
+ */
+const getPackageVersion = (
+    pkg: Package,
+    newVersion: string,
+    options: Record<'commitSha' | 'date' | 'deprecated' | 'preview', string | undefined>
+): string[] => {
+    let version: Array<string | undefined> = [newVersion];
+
+    if (pkg.deprecated) {
+        version.push(options.deprecated);
+    } else if (pkg.preview) {
+        version.push(options.preview);
+    }
+
+    return R.compact([...version, options.date, options.commitSha]);
+};
+
+run(async () => {
+    // Obtain the path of the repo root, useful for constructing absolute paths
+    const repoRoot = await gitRoot();
+
+    const [lernaFile, packageFile] = await Promise.all([
+        readJsonFile<Record<'lerna', string>>(path.join(repoRoot, 'lerna.json')),
+        readJsonFile<Package>(path.join(repoRoot, 'package.json')),
+    ]);
+
+    if (!lernaFile) {
+        return failure(1, 'lerna.json not found');
+    }
+
+    if (!packageFile) {
+        return failure(2, 'package.json not found');
```

Comment:
Why is this status code `2`? The [documentation](https://nodejs.org/dist/latest-v14.x/docs/api/process.html#process_exit_codes) indicates we should use `1` or perhaps `5`

Reply:
The documentation there is describing how Node uses exit codes when terminating programs, not necessarily prescribing their use. I'll use distinct ones that don't overlap (i.e. >13) though just to be extra careful.